lib/instrumentation/span.js

```diff
@@ -11,15 +11,20 @@ const TEST = process.env.ELASTIC_APM_TEST
 
 module.exports = Span
 
-function Span (transaction, name, type) {
+function Span (transaction, name, type, traceContext) {
   this.transaction = transaction
   this.ended = false
   this._db = null
   this._stackObj = null
   this._agent = transaction._agent
 
-  var current = this._agent._instrumentation.activeSpan || transaction
-  this.context = current.context.child()
+  // TODO: Should it be possible to give a traceparent string here? We probably don't need it though
+  if (traceContext) {
+    this.context = traceContext.child()
+  } else {
+    var current = this._agent._instrumentation.activeSpan || transaction
+    this.context = current.context.child()
+  }
```

Comment:
Maybe make spans _always_ expect a `TraceContext` instance to descend from, and move the `activeSpan` tracking stuff into the transaction instead?

Reply:
Not part of this PR anymore. Let's create a separate PR for that if we want to refactor it though.